### PR TITLE
Feature local storage

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Added local storage functionality
- Updated Index File
- Updated App.js File
- Added local storage functionality so that the user doesn't have to refill the form when page is refreshed

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -169,3 +169,26 @@ email.addEventListener("blur", () => {
...
 });
+
+/// STORING FORM DATA ///
+
+let nameSpace = document.getElementById("name");
+let textSpace = document.getElementById("text");
+function storeData() {
+  localStorage.setItem("UserName", nameSpace.value);
+  localStorage.setItem("Email", email.value);
+  localStorage.setItem("Text", textSpace.value);
```
Comment (ShahierNashaat) [inline app.js]: - [ ] Kindly notice that the form data for all inputs should be stored in a single object, and saved to a single local storage registry.

```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```

- [ ] Also you should use the `localStorage.getItem` and `localStorage.setItem` functions to save and retrieve the form data object to the local storage.

```
localStorage.setItem('contactFormInput', JSON.stringify(formData));
```

```
let savedFormData = localStorage.getItem('contactFormInput');
savedFormData = JSON.parse(savedFormData);
```

File: app.js [hunk 2]
```diff
@@ -169,3 +169,26 @@ email.addEventListener("blur", () => {
...
+
+function showData() {
+  let emailData = localStorage.getItem("Email");
+  let nameData = localStorage.getItem("UserName");
+  let textData = localStorage.getItem("Text");
+
+  email.value = emailData;
+  nameSpace.value = nameData;
+  textSpace.innerText = textData;
+}
```
Comment (ShahierNashaat) [inline app.js]: - [ ] Kindly notice that when the user loads the page, the code checks if there are any saved data in the local storage and if so the input fields are pre-filled with the data.

```javascript
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @IsmailChangezi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 3]
```diff
@@ -169,3 +169,46 @@ email.addEventListener("blur", () => {
...
+email.addEventListener("change", () => {
+  addToLocal();
+});
+
+textSpace.addEventListener("change", () => {
+  addToLocal();
+});
+
+if (formData === null) {
+  // console.log('Empty element');
```
Comment (Ismailco) [inline app.js:207]: - [ ] [OPTIONAL] Kindly avoid pushing commented code to your repository, it makes your code looks messy

**Review by Ismailco — APPROVED** [review]
### Hi @IsmailChangezi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Github Flow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**